### PR TITLE
Unify class definitions of common concepts and move them into node-commons package

### PR DESCRIPTION
**Pull request by xingtanzjr**
## Description

* Align class definitions of common concepts which will be used by many sub-projects.
* Move the common definition into `node-commons`
* Modify corresponding codes which leverage these common classes.

==== Update ====
Merge master and rewrite some classes for confignode part. The tests has passed.
image

### CODE REVIEW

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/ConsensusGroupId.java [hunk 1]
```diff
@@ -17,17 +17,17 @@
  * under the License.
  */
 
-package org.apache.iotdb.consensus.common;
+package org.apache.iotdb.commons.partition;
```
Comment (CRZbulabula) [inline node-commons/src/main/java/org/apache/iotdb/commons/partition/ConsensusGroupId.java:20]: Place the ConsensusGroupId in org.apache.iotdb.commons.consensus might be better.

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/Endpoint.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
  * under the License.
  */
 
-package org.apache.iotdb.consensus.common;
+package org.apache.iotdb.commons.partition;
```
Comment (CRZbulabula) [inline node-commons/src/main/java/org/apache/iotdb/commons/partition/Endpoint.java:20]: The Endpoint is not only used in partition module. For instance, the consensus layer also needs to use the Endpoint. So place the Endpoint to another package might be better. Like org.apache.iotdb.commons.tools?

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/GroupType.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
  * under the License.
  */
 
-package org.apache.iotdb.consensus.common;
+package org.apache.iotdb.commons.partition;
```
Comment (CRZbulabula) [inline node-commons/src/main/java/org/apache/iotdb/commons/partition/GroupType.java:20]: Place the GroupType in org.apache.iotdb.commons.consensus might be better.

**Review by CRZbulabula — CHANGES_REQUESTED** [review]
PTAL. Since this pr is fell behind the master. There was a prodigious construction in ConfigNode. So please merge master or cherry-pick [IOTDB-2684](https://github.com/apache/iotdb/pull/5374) first. The current modification in PartitionTable is useless, just discard it~

**Review by wangchao316 — CHANGES_REQUESTED** [review]
pull master ,  have some endpoint...

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/ConsensusGroupId.java (hunk 1 above)
Comment (xingtanzjr) [reply node-commons/src/main/java/org/apache/iotdb/commons/partition/ConsensusGroupId.java]: Done

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/GroupType.java (hunk 3 above)
Comment (xingtanzjr) [reply node-commons/src/main/java/org/apache/iotdb/commons/partition/GroupType.java]: Done

**Comment by xingtanzjr** [comment]
> pull master , have some endpoint...

Done

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/Endpoint.java (hunk 2 above)
Comment (xingtanzjr) [reply node-commons/src/main/java/org/apache/iotdb/commons/partition/Endpoint.java]: Done

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/RegionInfoPersistence.java [hunk 4]
```diff
@@ -156,8 +160,13 @@ public StorageGroupSchemaDataSet getStorageGroupSchema() {
                                   .getEndPoint());
                         }
                       });
-              schemaRegionReplicaSet.setSchemaRegionId(entity.getKey());
-              schemaRegionReplicaSet.setEndPointList(endPoints);
+              schemaRegionReplicaSet.setId(
+                  new ConsensusGroupId(GroupType.SchemaRegion, entity.getKey()));
+              // TODO: (xingtanzjr) We cannot get the dataNodeId here, use 0 as the placeholder
```
Comment (xingtanzjr) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/RegionInfoPersistence.java:165]: @CRZbulabula @wangchao316 , we cannot get the DataNodeId here, so I use `0` as a temporary placeholder... Please take a look

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/RegionInfoPersistence.java (hunk 4 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/RegionInfoPersistence.java:165]: That's OK, I'll fix it in next PR~

**Review by CRZbulabula — APPROVED** [review]
LGTM.

**Review by qiaojialin — APPROVED** [review]

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by CRZbulabula — APPROVED** [review]

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**